pig/src/org/partiql/pig/domain/model/TypeDomainSemanticChecker.kt

```diff
@@ -69,10 +69,11 @@ private class TypeDomainSemanticChecker(private val typeDomain: TypeDomain) {
                 }
                 DataType.Ion,
                 DataType.Int,
+                DataType.Bool,
                 DataType.Symbol -> {
                     /* do nothing, these are always valid */
                 }
-            }
+            }.let{}
```

Comment:
Is this `.let{}` required?

Reply:
Yes. It makes the when an expression which means Kotlin will give an error if there's a missing branch instead of a warning.

Reply:
It's not "required", however it makes that `when` an expression which means the Kotlin compiler will issue an error and not a warning if there is a missing branch.